runtime/doc/api-ui-events.txt

```diff
@@ -877,6 +878,24 @@ must handle.
 	    True if the message should be appeneded to the previous message,
 	    rather than started on a new line. Is set for |:echon|.
 
+	msg_id
+	    Unique identifier for the message in history. Is greater or equal
+	    to 1. 0 indicates msg_id was not assigned. When message of same id
+	    appears it should replace the older message.
+
+	extra_info
```

Comment:
Let's call this `progress`. All the fields are for that purpose. If we need a free-form field (`data`) we can always add it later.